org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServerWrapper.java

```diff
@@ -369,6 +370,27 @@ private URI getRootURI() {
 		return null;
 	}
 
+	@Nullable
+	public String getCommandLine() {
+		if (lspStreamProvider instanceof ProcessInfo pi) {
+			return pi.getCommandLine();
+		}
+		return null;
+	}
```

Comment:
Those utility method are not best in this class; have something that returns (or better, some adapter) to get the ProcessInfo (or better ProcessHandle) is enough. Then consumers can dereference it as needed.

Reply:
ok, I am addressing this separately with #755